### DISCUSSION TITLE
[v3] Session regeneration problem (419 error)

### DISCUSSION THREAD

**Discussion by beholdr** [livewire/livewire#6676] [1. Help] [answered]
I have an authorization form (Livewire component) which is calling `Session::regenerate();` after user is authorized. I want this to work without page reload because there are other LW-components on the page. Problem is, after login I always got an alert «The page has expired» (419 error). This is because CSRF token has changed after session regeneration.

With Livewire v2 I solved this issue by sending custom event on authorization with new CSRF token. And on JS side I listen for this event and substitute old CSRF token with the new value like this:

``` js
Livewire.on('sessionRegenerated', (csrfToken) => {
    window.livewire_token = csrfToken
})
```

And it works fine. But with v3 it's not working anymore. CSRF token now stored in the `window.livewireScriptConfig.csrf` variable. But even when I update this value in my listener, requests still use old token because it’s cached in the `csrf` variable set in [getCsrfToken](https://github.com/livewire/livewire/blob/2118197070d1ae5bd689615387ef545ed5d7d697/js/utils.js#L168) function after first access.

Is there any way to refresh CSRF token without full page reload? I could disable session regeneration on auth, but I don’t think it’s a recommended practice.

**Comment by yilanboy** [comment, marked as answer]
I saw Caleb said this will be fixed in the next release.

see #6784

**Reply by beholdr** [reply]
Thanks for pointing it out.
